### PR TITLE
feat: allow build within docker container

### PR DESCRIPTION
**Pull request by cbernat**
- Add docker file for builds.
- Use dockerfile from build-n-publish script

Output: cdn folder as we used (except for development enviroment)
* production --> buildxxx (https://cdn.fromdoppler.com/doppler-webapp/build15323/#/login)
* int --> int-buildxxx (https://cdn.fromdoppler.com/doppler-webapp/int-build1532/#/login)
* qa --> qa-buildxxx --> (https://cdn.fromdoppler.com/doppler-webapp/qa-build1532/#/login)
* development --> development-buildxxx (before it was dev-buildxxx) (https://cdn.fromdoppler.com/doppler-webapp/development-build1532/#/login)

### CODE REVIEW

**Comment by cbernat** [comment]
Test this please.

File: build-n-publish.sh [hunk 1]
```diff
@@ -3,64 +3,60 @@
...
-    -e AKAMAI_CDN_HOSTNAME \
-    -e AKAMAI_CDN_USERNAME \
-    -e AKAMAI_CDN_PASSWORD \
-    -e AKAMAI_CDN_CPCODE \
-    -e "PROJECT_NAME=$pkgName" \
-    -e "VERSION_NAME=$pkgVersionQa" \
-    -v `pwd`/build:/source \
-    dopplerrelay/doppler-relay-akamai-publish
+# TODO: It could break concurrent deployments with different docker accounts
+docker login -u="$DOCKER_WEBAPP_USERNAME" -p="$DOCKER_WEBAPP_PASSWORD"
```
Comment (andresmoschini) [inline build-n-publish.sh]: I think that it is not necessary because we do not upload anything to Docker Hub.

File: build-n-publish.sh [hunk 2]
```diff
@@ -3,64 +3,60 @@
...
+    then  
+        env_version=$pkgVersion;
+    else
+        env_version=$environment-$pkgVersion;
+    fi
+
+    echo deploying into $env_version folder ...
+
+    docker build --pull \
+        -t darosw/doppler-webapp:$environment \
```
Comment (andresmoschini) [inline build-n-publish.sh]: I think that we do not want to tag this image.

File: build-n-publish.sh [hunk 3]
```diff
@@ -3,64 +3,60 @@
...
+        --build-arg environment=$environment \
+        --build-arg env_version=$env_version \
+        --build-arg cdn_hostname=$AKAMAI_CDN_HOSTNAME \
+        --build-arg cdn_username=$AKAMAI_CDN_USERNAME \
+        --build-arg cdn_password=$AKAMAI_CDN_PASSWORD \
+        --build-arg cdn_cpcode=$AKAMAI_CDN_CPCODE \
+        -f Dockerfile.BUILDS \
```
Comment (andresmoschini) [inline build-n-publish.sh]: What do you think about: 

```suggestion
        -f Dockerfile.BUILDS_AND_CDN \
```

Or



```suggestion
        -f Dockerfile.PULLREQUESTS \
```

**Comment by andresmoschini** [comment]
test this please

**Comment by andresmoschini** [comment]
WOW! Impressive results! 

It not only fixes the issue with folders that Jenkins cannot remove but it also, as the collateral effect, in most of the cases, optimizes de build time:

image

We need to apply this same approach in:

* UI Library (Probably @IgnacioRodrigues will work on that)
* Shopify (@javierburger? @CarlosSampedro? or me)
* Datahub (@VicThorX? @elsupergomez? or @leoslopez)
* MSEditor (@mvillaseco? @lucianaDilizio?)
* MSForms (@juanasas? @CarlosPintos?)

File: build-n-publish.sh (hunk 3 above)
Comment (cbernat) [reply build-n-publish.sh]: Sounds nice.

**Comment by cbernat** [comment]
test this please

File: build-w-docker.sh [hunk 4]
```diff
@@ -1,28 +0,0 @@
...
-
-docker run --rm \
-    -v `pwd`:/work \
-    -w /work \
-    node:10 \
-    /bin/sh -c "\
-        yarn \
-        && yarn run build:$environment \
-        && printf \"$cdnBaseUrl\n$environment\n$pkgVersion\n$versionFull\nbuildNo$pkgBuild\n$pkgCommitId\" > build/version.txt \
-    "
```
Comment (andresmoschini) [inline build-w-docker.sh]: I think that this file could be still be useful to build locally (in my environment I have Docker but not Node)

Comment (cbernat) [reply build-w-docker.sh]: Added this in dockerfile.

Comment (cbernat) [reply build-w-docker.sh]: I will resto this file then 😄

**Comment by andresmoschini** [comment]
🚀

**Pull request merged**